### DISCUSSION TITLE
Is there any specification for the test result format?

### DISCUSSION THREAD

**Discussion by Martin91** [allure-framework/allure2#1764] [General Discussion]
Hi, I need to develop an adapter for our own test framework, but I couldn't find any document about the format of the test result JSON files.

**Comment by baev** [comment]
there is no documentation yet, but we have a SDK's for each language, so you can use it. Or you can look how it's done here https://github.com/allure-framework/allure-js/blob/master/packages/allure-js-commons/src/model.ts

**Reply by Martin91** [reply]
Thanks! It helps so much!

**Comment by michaelpj** [comment]
Just wanted to re-up this: I was thinking about writing some test report generators for some Haskell test frameworks, but it's difficult without any documentation of the test format. It would be superb if it was in a machine-readable format such that the code for working with it could be generated :)

**Reply by baev** [reply]
Can you share some examples of format you have in mind?

**Reply by michaelpj** [reply]
A JSON Schema for the format might work, for example. Many languages have code generators that will generate code for working with a format from JSON Schema. Maybe you could even use it to simplify some of your existing model definitions!

**Comment by baev** [comment]
The testresult.json file documentation: https://allurereport.org/docs/how-it-works-test-result-file/
